data/data/vsphere/bootstrap/main.tf

```diff
@@ -42,7 +42,13 @@ resource "vsphere_virtual_machine" "vm_bootstrap" {
     template_uuid = var.template[0].uuid
   }
 
-  extra_config = {
+  extra_config = length(var.vsphere_bootstrap_network_kargs) > 0 ? {
```

Comment:
Instead of duplicating the map...

```
locals {
  extra_config_required = {
    "guestinfo.ignition.config.data"           = base64encode(var.ignition_bootstrap)
    "guestinfo.ignition.config.data.encoding"  = "base64"
    "guestinfo.hostname"                       = "${var.cluster_id}-bootstrap"
    "stealclock.enable"                        = "TRUE"
  }
  extra_config_optional = var.vsphere_bootstrap_network_kargs == "" ? {} : {"guestinfo.afterburn.initrd.network-kargs" = "${var.vsphere_bootstrap_network_kargs}"}

  extra_config = merge(local.extra_config_required, local.extra_config_optional) 
}
```

Reply:
omg, didnt know you could do that :cries:

Reply:
I tried the change and ended up getting an error:
`ERROR   on main.tf line 45, in resource "vsphere_virtual_machine" "vm_bootstrap": 
ERROR   45:   locals {                             
ERROR                                              
ERROR The block type name "locals" is reserved for use by Terraform in a future 
ERROR version.  `